docs/configuration/http_conn_man/http_conn_man.rst

```diff
@@ -97,6 +98,27 @@ http_codec_options
   <config_cluster_manager_cluster_http_codec_options>` option. See the comment there about
   disabling HTTP/2 header compression.
 
+.. _config_http_conn_man_http2_settings:
+
+http2_settings
+  *(optional, object)* Additional HTTP/2 codec options that are passed directly to the HTTP/2 codec. 
+  Currently supported settings are:
+
+  max_concurrent_streams
+    *(optional, integer)* `Maximum concurrent streams`_ allowed on one HTTP/2 connection. Valid values
+    range from 1 to 536870912 (2^29) and defaults to 1024.
+
+  initial_window_size
+    *(optional, integer)* `Initial flow-control window`_ size. Valid values range from 65535
```

Comment:
65535 is the initial value defined in the HTTP/2 spec. 

from http://httpwg.org/specs/rfc7540.html#rfc.section.6.9.3 ,
my understanding is that client are allowed to send up to initial window bytes before got peer's window setting. if it received a windows setting smaller, it will blocks until another WINDOW_UPDATE.
so it needs to handle WINDOW_UPDATE properly to avoid this, i am not sure if this is done in envoy/nghttp2 already, so to be safe, i don't allow value below the initial size.